# (APPENDIX) Appendix {-}
# Build TxDB package from GTF {#appendixA-buildTxDB}

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

To build a Bioconductor TxDB, you can follow these steps:

1. Start by converting the annotation GTF (e.g. Gencode V35) file you downloaded into a `GRange` instance.
2. Next, prepare metadata and convert the `GRange` instance into a `TxDB` package. 
3. Optionally, you can include gene annotation information (such as gene_name, gene_type, gene_ID, and gene_type) as a DataFrame instance in the data folder of the package. However, this step is not necessary if you're building an Ensembl DB package (EnsDb).

```{r build-gencodev35-TxDB, eval=FALSE}
library(rtracklayer)
library(GenomicFeatures)

## Define the destination and package name of your TxDB package
pkg_name <- "hg38.HomoSapiens.Gencode.v35"
dest_dir <- "/fh/fast/tapscott_s/CompBio/hg38"

## Where is my GTF file
gtf_file <- "/fh/fast/tapscott_s/CompBio/genome_reference/GRCh38/Annotation/gencode.v35.annotation.gtf"

## Import the GTF file into a GRange instance
gencode <- rtracklayer::import.gff(gtf_file)

## Define metadata: version, source, and etc.
organism <- "human"
release <- "v35"
dataSource <- paste0("ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_",
                     organism, "/", release)
metadata <- data.frame(
  name=c("Organism", "Resource URL", "Resource GTF file",
         "Taxonomy ID", "miRBase build ID", "Data source"),
  value=c("Homo sapiens", dataSource, gtf_file, NA, NA, dataSource))

## Prepare the metadata
metadata <- GenomicFeatures:::.prepareGFFMetadata(gtf_file, dataSource,
                                                  organism="Homo sapiens")

## Combine the GRange instance and metadata into a TxDB instance
txdb <- GenomicFeatures:::makeTxDbFromGRanges(gr=gencode,
                                              metadata=metadata)

## Build a TxDb package
makeTxDbPackage(txdb, version="3.6.2", author="Chao-Jen Wong",
                pkgname=pkg_name, destDir=dest_dir, license="Artistic-2.0",
                provider="Gencode", providerVersion=release,
                maintainer="Chao-Jen Wong <cwon2@fredhutch.org>")
```

The downside of the `TxDB` package is that it does not carry the corresponding columns of gene_name, gene_type, and gene_symbols as the original GTF file obtained. So adding a `DataFrame` instance to include that information might provide convenience in future usage. 

```{r gene-annotation, eval=FALSE}
gene.anno <- mcols(gencode)[gencode$type=="gene", ]
rownames(gene.anno) <- gene.anno$gene_id

#' put annotation to the data directory
data_dir <- file.path(dest_dir, pkg_name, "data")
if (!file.exists(data_dir))
    dir.create(data_dir)
save(gene.anno, file=file.path(data_dir, "gene.anno.rda"))
```

### Build EnsDb package using `AnnotationHub`
In retrospect, I would use `AnnotationHub()` and `GenomicFeatures::makeEnsembldbPackage()` to make an `EnsDB` package instead of `TxDB` because `EnsDB` has slots/functions to retrieve the gene information. Below is an example: 
```{r annotaitonHub-example, eval=FALSE}
#'
#' EnsDb.Hsapiens.v92: 
#'
library(AnnotationHub)
library(GenomicFeatures)
ah <- AnnotationHub()
query(ah, c("hsapiens"))
edb <- ah[["AH60977"]]
seqlevelsStyle(edb) <- "NCBI"
makeEnsembldbPackage(ensdb=dbfile(dbconn(edb)), version="1.0.0",
                     maintainer="Chao-Jen Wong <cwon2@fredhutch.org>",
                     author="Chao-Jen Wong",
                     destDir="/fh/fast//tapscott_s/CompBio/hg38")
```
