--- 
title: "Wellstone bilateral tibialis anterior FSHD biopsied muscle study"
author: "Chao-Jen Wong"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
cover-image: images/cover.jpeg
url: https://fredHutch.github.io/Wellstone_BiLateral_Biopsy
description: |
  This book is written to support computational transparency and reproducibility for our publication titled "Validation study of the association of MRI and FSHD gene signature reveals markers of whole muscle and systemic disease progression".
biblio-style: apalike
csl: chicago-fullnote-bibliography.csl
---

# About

The primary objective of this book is to advocate for computational transparency and reproducibility in conjunction with our manuscript entitled ["Validation study of the association of MRI and FSHD gene signature reveals markers of whole muscle and systemic disease progression."](https://www.biorxiv.org/content/biorxiv/early/2023/02/20/2023.02.20.529303.full.pdf) By offering detailed explanations of our decision-making processes grounded in machine learning and statistical inferences, we provide readers with a comprehensive comprehension of our computational methods and coding. Furthermore, the book includes reproducible codes that generate real-time results, enhancing the accessibility and usability of our research.

In this study, we obtained bilateral tibialis anterior (TA) biopsied muscles from 34 subjects. The muscle biopsies (n=68) were processed for MRI characterization, histopathological scoring, RNA extraction for RNA sequencing , and DNA extraction for bisulfite sequencing of the 4qA permissive allele in the region of exon 2-3. Through the metrics yield by these processes, we were able to validate the association of MRI and RNA molecular signatures, and identify immune cell markers associated with immune cell infiltrates and symmetric trends in the disease. Additionally, we curated and proposed panels of genes as post-therapeutic targets for clinical trial designs.

This book discusses the following topics:

1. Transcript-based assessment of muscle cell type content of the FSHD biopsied muscle
2. FSHD molecular signatures, including DUX4, extracurricular matrix, inflammatory, complement activation, and immunoglobulins
3. ML classification based on the FSHD molecular signatures
4. Identification of six baskets containing genes that exhibit specific signatures associated with FSHD
5. Association of MRI characteristics with signatures of DUX4 expression 
6. Bilateral comparison analysis
7. Verification of immune cell infiltrate signatures and immune cell type proportions


## Software

- R (4.2.2) and packages from the Bioconductor (3.17) and Tidyverse projects. Most used packages include `GenomicAlignments`, `GenomicFeatures`, `DESeq2`, `dplyr`, `ggplot2`, `caret`, and genomic-related BioC packages for annotation.
- `fastqc`, `Subread`, `samTools`, and `cutadapt` for RNA-seq data preprocessing, and `PLIER` for immune cell type proportions.

## Repos structure
The section outlines the structure of our repository, which is available at [here](https://github.com/FredHutch/Wellstone_BiLateral_Biopsy). 

```
\data: 
    |- sanitized.dds.rda: DESeqDataSet instance; longitudinal RNA-seq
    |- longitudinal_dds.rda: DESeqDataSet instance include the longitudinal 
       RNA-seq gene counts and metadata (MRI, pathology, and clinical data)
    |- cluster_df.rda: data.frame instance
    |- bilateral_dds.rda: DESeqDataSet instance; bilateral RNA-seq data
    |- comprehensive_df.rda: data.frame instance obtaining the biletarl
       cohort's clinical, MRI, pathology, FSHD molecular signature scores,
       and methylation data

\docs: *.html; folder hosts the gitbook 
\gitbook: *.Rmd that makes the gitbook
\scripts: *.R; un-orgnaized R code of our initial data 
          exploration and bioinformatics analysis
\extdata: *.xlsx; extra files and supplemental tables
```

## Annotation package
To conducto our RNA-seq analysis, used annotations from Gencode v35 and built a customized `TxDB` package, which we named `hg38.HomoSapiens.Gencode.v35`. For instructions on building Bioconductor `TxDB` or `EsbDB` annotation packages, please refer to Appendix A.


## RNA-seq preprocessing and gene counts
To pre-preprocess the RNA-seq data, we filtered out unqualified raw reads, trimmed the Illumina universal adapters using _Trimmomatic_, and aligned the remaining reads to GRCh38.p13 with _Rsubread_.

The code chunk below provides an example of how we used `GenomicAlignments`, `BiocParallel`, and a customized `TxDB` package (`hg38.HomoSapiens.Gencode.v35`) to count reads and generate an `RangedSummarizedExperiment` object. Note that the `sort_files` parameter indicates the location of the sorted, indexed BAM files used in our analysis. In the `summarizOverlaps()` function, I chose to use `Intersectionstrict` mode that only count reads that are completely contained within the range of exons, and ignore any ambiguous reads that straddle different gene features. 

```{r make-se, eval=FALSE}
# sort_files gives the location of my bam files
sort_files <- list.files(scratch_dir, full.name=TRUE, pattern="\\.bam$")
library(GenomicFeatures)
library(GenomicAlignments)
library(Rsamtools)
library(BiocParallel)
library(hg38.HomoSapiens.Gencode.v35)
bp_param <- BiocParallel::MulticoreParam(workers=12L)
register(bp_param)

features <- GenomicFeatures::exonsBy(hg38.HomoSapiens.Gencode.v35, by="gene")
features <- keepStandardChromosomes(features,
                                    species = "Homo_sapiens",
                                    pruning.mode="fine")
rse <- 
  GenomicAlignments::summarizeOverlaps(features=features,
                                       reads =
                                         Rsamtools::BamFileList(sort_file),
                                       mode = "IntersectionStrict",
                                       ignore.strand=TRUE, 
                                       singleEnd=TRUE,
                                       inter.feature=TRUE, 
                                       BPPARAM=bp_param)
```

