<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 1 About | Wellstone bilateral tibialis anterior FSHD biopsied muscle study</title>
<meta name="author" content="Chao-Jen Wong">
<meta name="description" content="This book supports computational transparency and reproducibility for our publication titled “Validation study of the association of MRI and FSHD gene signature reveals markers of whole muscle and systemic disease progression”.">
<meta name="generator" content="bookdown 0.32 with bs4_book()">
<meta property="og:title" content="Chapter 1 About | Wellstone bilateral tibialis anterior FSHD biopsied muscle study">
<meta property="og:type" content="book">
<meta property="og:url" content="https://fredHutch.github.io/Wellstone_BiLateral_Biopsy">
<meta property="og:image" content="https://fredHutch.github.io/Wellstone_BiLateral_Biopsy/images/cover.jpeg">
<meta property="og:description" content="This book supports computational transparency and reproducibility for our publication titled “Validation study of the association of MRI and FSHD gene signature reveals markers of whole muscle and systemic disease progression”.">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 1 About | Wellstone bilateral tibialis anterior FSHD biopsied muscle study">
<meta name="twitter:description" content="This book supports computational transparency and reproducibility for our publication titled “Validation study of the association of MRI and FSHD gene signature reveals markers of whole muscle and systemic disease progression”.">
<meta name="twitter:image" content="https://fredHutch.github.io/Wellstone_BiLateral_Biopsy/images/cover.jpeg">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Sen-0.4.5/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/Bitter-0.4.5/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Wellstone bilateral tibialis anterior FSHD biopsied muscle study</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="active" href="index.html"><span class="header-section-number">1</span> About</a></li>
<li><a class="" href="study-cohort.html"><span class="header-section-number">2</span> Study cohort</a></li>
<li><a class="" href="muscle-content.html"><span class="header-section-number">3</span> Transcript-based accessment for muscle content</a></li>
<li><a class="" href="MRI-vs-DUX4-score.html"><span class="header-section-number">4</span> MRI characteristics and DUX4 molecular signatures</a></li>
<li><a class="" href="DUX4-basket.html"><span class="header-section-number">5</span> DUX4 basket</a></li>
<li><a class="" href="other-baskets.html"><span class="header-section-number">6</span> Inflammatory, ECM, complement activation and immunoglobulins signatures</a></li>
<li><a class="" href="bilateral-comparison.html"><span class="header-section-number">7</span> Bilateral comparison</a></li>
<li><a class="" href="immune-cell-infiltrates.html"><span class="header-section-number">8</span> Immune cell infiltrates signature</a></li>
<li><a class="" href="plier.html"><span class="header-section-number">9</span> Immune cell type proportion by PLIER</a></li>
<li><a class="" href="complement-scoring.html"><span class="header-section-number">10</span> Complement scoring</a></li>
<li><a class="" href="dna-methylation.html"><span class="header-section-number">11</span> DNA methylation of FSHD D4Z4 allele</a></li>
<li><a class="" href="references.html">References</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="appendixA-buildTxDB.html"><span class="header-section-number">A</span> Build TxDB package from GTF</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/FredHutch/Wellstone_BiLateral_Biopsy">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><!--bookdown:title:end--><!--bookdown:title:start--><div id="about" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> About<a class="anchor" aria-label="anchor" href="#about"><i class="fas fa-link"></i></a>
</h1>
<p>The objective of this book is to advocate for computational transparency and reproducibility of our FSHD bilateral cohort study, titled <a href="https://www.biorxiv.org/content/biorxiv/early/2023/02/20/2023.02.20.529303.full.pdf">“Validation study of the association of MRI and FSHD gene signature reveals markers of whole muscle and systemic disease progression.”</a> This book includes detailed explanations of our decision-making processes grounded in bioinformatics analysis, machine learning, and statistical inferences along with reproducible codes (in R) that generate results and figures on the fly.</p>
<p>In this FSHD bilateral cohort study, we recruited 34 FSHD patients and obtained their left and right tibialis anteria (TA) muscle biopsies. On the biopsies, we performed MRI characterization, histopathological scoring, RNA extraction for RNA sequencing, and DNA extraction for bisulfite sequencing of the 4qA permissive allele in the region of exon 2-3. Through the metrics yield by these processes, we studied the following topics:</p>
<ol style="list-style-type: decimal">
<li>Transcript-based assessment of muscle cell type content of the FSHD biopsied muscle</li>
<li>FSHD molecular signatures, including DUX4, extracurricular matrix, inflammatory, complement activation, and immunoglobulins</li>
<li>ML classification based on the FSHD molecular signatures</li>
<li>Identification of six baskets containing genes that exhibit specific signatures associated with FSHD</li>
<li>Association of MRI characteristics with signatures of DUX4 expression</li>
<li>Bilateral comparison analysis</li>
<li>Verification of immune cell infiltrate signatures and immune cell type proportions</li>
</ol>
<div id="software" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Software<a class="anchor" aria-label="anchor" href="#software"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>R (4.2.2) and packages from the Bioconductor (3.17) and Tidyverse projects. Most used packages include <code>GenomicAlignments</code>, <code>GenomicFeatures</code>, <code>DESeq2</code>, <code>dplyr</code>, <code>ggplot2</code>, <code>caret</code>, and genomic-related BioC packages for annotation.</li>
<li>
<code>fastqc</code>, <code>Subread</code>, <code>samTools</code>, and <code>cutadapt</code> for RNA-seq data preprocessing, and <code>PLIER</code> for immune cell type proportions.</li>
</ul>
</div>
<div id="repos-structure" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Repos structure<a class="anchor" aria-label="anchor" href="#repos-structure"><i class="fas fa-link"></i></a>
</h2>
<p>The section outlines the structure of our repository, which is available at <a href="https://github.com/FredHutch/Wellstone_BiLateral_Biopsy">here</a>.</p>
<pre><code>\data: 
    |- sanitized.dds.rda: DESeqDataSet instance; longitudinal RNA-seq
    |- longitudinal_dds.rda: DESeqDataSet instance include the longitudinal 
       RNA-seq gene counts and metadata (MRI, pathology, and clinical data)
    |- cluster_df.rda: data.frame instance
    |- bilateral_dds.rda: DESeqDataSet instance; bilateral RNA-seq data
    |- comprehensive_df.rda: data.frame instance obtaining the biletarl
       cohort's clinical, MRI, pathology, FSHD molecular signature scores,
       and methylation data

\docs: *.html; folder hosts the gitbook 
\gitbook: *.Rmd that makes the gitbook
\scripts: *.R; un-orgnaized R code of our initial data 
          exploration and bioinformatics analysis
\extdata: *.xlsx; extra files and supplemental tables</code></pre>
</div>
<div id="annotation-and-txdb-package" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> Annotation and TxDB package<a class="anchor" aria-label="anchor" href="#annotation-and-txdb-package"><i class="fas fa-link"></i></a>
</h2>
<p>The gene annotation for our RNA-seq analysis is based on Gencode version 35. We built a Bioconductor <code>TxDB</code> package named <code>hg38.HomoSapiens.Gencode.v35</code> to facilitate gene counts. See Appendix A on how to build Bioconductor <code>TxDB</code> or <code>EsbDB</code> annotation packages (add cross-reference here).</p>
</div>
<div id="rna-seq-preprocessing-and-gene-counts" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> RNA-seq preprocessing and gene counts<a class="anchor" aria-label="anchor" href="#rna-seq-preprocessing-and-gene-counts"><i class="fas fa-link"></i></a>
</h2>
<p>The pre-preprocessing of RNA-seq data, we filtered out unqualified raw reads, trimmed the Illumina universal adapters using <em>Trimmomatic</em>, and aligned the remaining reads to GRCh38.p13 with <em>Rsubread</em>.</p>
<p>The code chunk below provides an example of how we used <code>GenomicAlignments</code>, <code>BiocParallel</code>, and a customized <code>TxDB</code> package (<code>hg38.HomoSapiens.Gencode.v35</code>) to count reads and generate an <code>RangedSummarizedExperiment</code> object. Note that the <code>sort_files</code> parameter indicates the location of the sorted, indexed BAM files used in our analysis. In the <code>summarizOverlaps()</code> function, I chose to use <code>Intersectionstrict</code> mode that only count reads that are completely contained within the range of exons, and ignore any ambiguous reads that straddle different gene features.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># sort_files gives the location of my bam files</span></span>
<span><span class="va">sort_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">scratch_dir</span>, full.name<span class="op">=</span><span class="cn">TRUE</span>, pattern<span class="op">=</span><span class="st">"\\.bam$"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures">GenomicFeatures</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicAlignments">GenomicAlignments</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Rsamtools">Rsamtools</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel">BiocParallel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">hg38.HomoSapiens.Gencode.v35</span><span class="op">)</span></span>
<span><span class="va">bp_param</span> <span class="op">&lt;-</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html">MulticoreParam</a></span><span class="op">(</span>workers<span class="op">=</span><span class="fl">12L</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html">register</a></span><span class="op">(</span><span class="va">bp_param</span><span class="op">)</span></span>
<span></span>
<span><span class="va">features</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html">exonsBy</a></span><span class="op">(</span><span class="va">hg38.HomoSapiens.Gencode.v35</span>, by<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html">keepStandardChromosomes</a></span><span class="op">(</span><span class="va">features</span>,</span>
<span>                                    species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>,</span>
<span>                                    pruning.mode<span class="op">=</span><span class="st">"fine"</span><span class="op">)</span></span>
<span><span class="va">rse</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">GenomicAlignments</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicAlignments/man/summarizeOverlaps-methods.html">summarizeOverlaps</a></span><span class="op">(</span>features<span class="op">=</span><span class="va">features</span>,</span>
<span>                                       reads <span class="op">=</span></span>
<span>                                         <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html">BamFileList</a></span><span class="op">(</span><span class="va">sort_file</span><span class="op">)</span>,</span>
<span>                                       mode <span class="op">=</span> <span class="st">"IntersectionStrict"</span>,</span>
<span>                                       ignore.strand<span class="op">=</span><span class="cn">TRUE</span>, </span>
<span>                                       singleEnd<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                                       inter.feature<span class="op">=</span><span class="cn">TRUE</span>, </span>
<span>                                       BPPARAM<span class="op">=</span><span class="va">bp_param</span><span class="op">)</span></span></code></pre></div>

</div>
</div>
  <div class="chapter-nav">
<div class="empty"></div>
<div class="next"><a href="study-cohort.html"><span class="header-section-number">2</span> Study cohort</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#about"><span class="header-section-number">1</span> About</a></li>
<li><a class="nav-link" href="#software"><span class="header-section-number">1.1</span> Software</a></li>
<li><a class="nav-link" href="#repos-structure"><span class="header-section-number">1.2</span> Repos structure</a></li>
<li><a class="nav-link" href="#annotation-and-txdb-package"><span class="header-section-number">1.3</span> Annotation and TxDB package</a></li>
<li><a class="nav-link" href="#rna-seq-preprocessing-and-gene-counts"><span class="header-section-number">1.4</span> RNA-seq preprocessing and gene counts</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/FredHutch/Wellstone_BiLateral_Biopsy/blob/master/gitbook/index.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/FredHutch/Wellstone_BiLateral_Biopsy/edit/master/gitbook/index.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Wellstone bilateral tibialis anterior FSHD biopsied muscle study</strong>" was written by Chao-Jen Wong. It was last built on 2023-12-16.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
